core/src/wallet/ripple/explorers/NodeRippleLikeBlockchainExplorer.cpp

```diff
@@ -154,18 +154,17 @@ namespace ledger {
                             !json["result"].IsObject()) {
                             throw make_exception(api::ErrorCode::HTTP_ERROR, "Failed to broadcast transaction, no (or malformed) field \"result\" in response");
                         }
-                        //Is there an tx_json field ?
+
                         auto resultObj = json["result"].GetObject();
-                        if (!resultObj.HasMember("tx_json") || !resultObj["tx_json"].IsObject()) {
-                            throw make_exception(api::ErrorCode::HTTP_ERROR, "Failed to broadcast transaction, no (or malformed) field \"tx_json\" in response");
-                        }
 
-                        //Is there an hash field ?
-                        auto jsonObj = resultObj["tx_json"].GetObject();
-                        if (!jsonObj.HasMember("hash") || !jsonObj["hash"].IsString()) {
-                            throw make_exception(api::ErrorCode::HTTP_ERROR, "Failed to broadcast transaction, no (or malformed) field \"hash\" in response");
+                        if (resultObj.HasMember("engine_result") && resultObj["engine_result"] == "tesSUCCESS") {
+                          auto txnObj = resultObj["tx_json"].GetObject();
```

Comment:
You forgot to test whether `"tx_json"` is in `resultObj`. :wink:

Reply:
I intentionally removed those checks, because if the status is `tesSUCCESS`, then Ripple's documentation provides us with the guarantee of a predictable JSON structure.

If you think it's not a good practice, then I'll add the checks.

Reply:
I get your point, but the JSON is broacasted on a network and could be tampered with. Or a node might have a bug and provide an ill-formed JSON data.

I always take the side of “everything that comes from the user, the network, or any other service” is not trustworthy and should be checked. The problem with such assumptions is that it’s very easy to break our lib by providing a JSON structure sligthly different. Instead of crashing the app, we should emit en error stating that a key is missing.

Reply:
Done. 👍